File: app/src/main/java/io/github/gusandrianos/foxforreddit/data/models/RedditVideoPreview.java
package io.github.gusandrianos.foxforreddit.data.models;

import com.google.gson.annotations.SerializedName;

public class RedditVideoPreview{

    @SerializedName("duration")
    private int duration;

    @SerializedName("is_gif")
    private boolean isGif;

    @SerializedName("dash_url")
    private String dashUrl;

    @SerializedName("fallback_url")
    private String fallbackUrl;

    @SerializedName("width")
    private int width;

    @SerializedName("scrubber_media_url")
    private String scrubberMediaUrl;

    @SerializedName("hls_url")
    private String hlsUrl;

    @SerializedName("transcoding_status")
    private String transcodingStatus;

    @SerializedName("height")
    private int height;
}

File: app/src/main/java/io/github/gusandrianos/foxforreddit/data/models/RedditVideoPreview.kt
package io.github.gusandrianos.foxforreddit.data.models

import com.google.gson.annotations.SerializedName

class RedditVideoPreview {
    @SerializedName("duration")
    val duration = 0

    @SerializedName("is_gif")
    val isGif = false

    @SerializedName("dash_url")
    val dashUrl: String? = null

    @SerializedName("fallback_url")
    val fallbackUrl: String? = null

    @SerializedName("width")
    val width = 0

    @SerializedName("scrubber_media_url")
    val scrubberMediaUrl: String? = null

    @SerializedName("hls_url")
    val hlsUrl: String? = null

    @SerializedName("transcoding_status")
    val transcodingStatus: String? = null

    @SerializedName("height")
    val height = 0
}